### DISCUSSION TITLE
Copying text inside a document field component block copies the whole component block

### DISCUSSION THREAD

**Discussion by PhilBookst** [keystonejs/keystone#7507] [Questions] [answered]
Hey Keystone Team,

When copying text inside a component block, keystone copies the whole component block structure instead of just the text.
This is really annoying when you start working on big texts and you cannot copy the text inside the blocks.

Is this the desired behaviour? If so, is there a way to opt-out of it? 

https://user-images.githubusercontent.com/68562883/166098469-ba8529aa-148b-4c40-a069-2747f463df6e.mp4

**Comment by fatihsolhan** [comment, marked as answer]
Hi @PhilBookst 

I think it is intended behavior. Think of it as copying **bold** styled text and pasting it again.
To paste without formatting, you can use <kbd>Ctrl + Shift + V</kbd>

**Reply by PhilBookst** [reply]
Hey @fatihsolhan  

Yes for text formatting like **bold** / _italic_ or heading settings it makes sense, but I think there are not many situations you'd want to copy the whole component block.

Our writers say it's more cumbersome than helpful, since if you'd want the component block it's just 1 click away but if you need to refactor some text from a component block in the normal text area it just causes problems and more work.

Maybe it's possible to introduce a boolean flag in the component block which allows copying the whole component or just the text within it?

And thanks for the suggestion with `Ctrl/Cmd + Shift + V` that's a good workaround, I hope our writers can adapt this method.

**Reply by fatihsolhan** [reply]
That's a good point actually, component blocks should be copied as plain text since it is just 1 click away as you said

I find another workaround if you want;
writers can also use this [chrome extension](https://chrome.google.com/webstore/detail/copy-as-plain-text/eneajgkmdhmjmloiabgkpkiooaejmlpk/related?hl=en) to copy plain text by right click instead of a keyboard shortcut. 
Still, one extra work to just copy a text... 😄

**Reply by PhilBookst** [reply]
Ok I got it fixed by adding `onCopy={e => e.stopPropagation()}` to the element wrapping the keystone child field
